extendr-macros/src/wrappers.rs

```diff
@@ -354,7 +354,16 @@ pub fn translate_formal(input: &FnArg, self_ty: Option<&syn::Type>) -> syn::Resu
         // function argument.
         FnArg::Typed(ref pattype) => {
             let pat = &pattype.pat.as_ref();
-            Ok(parse_quote! { #pat : extendr_api::SEXP })
+            let pat_ident = match pat {
+                syn::Pat::Ident(ref pat_ident) => &pat_ident.ident,
+                _ => {
+                    return Err(syn::Error::new_spanned(
+                        input,
+                        "failed to translate name of argument",
+                    ))
+                }
+            };
+            Ok(parse_quote! { #pat_ident : extendr_api::SEXP })
```

Comment:
Does this catch errors where previously, they were swallowed by the `parse_quote!` macro?

Reply:
I don't think so. It is hard to reason about this. But basically, I think it is because parsing this requires handling all the cases.. but here I don't think there are any other cases. I think `mut arg1` was simply being plastered everywhere, instead of `arg1` being extracted and used everywhere, and then `mut` just being amended when needed..